File: server/src/main/java/ua/com/alevel/service/educationalprocess/ReportService.java
package ua.com.alevel.service.educationalprocess;

import org.springframework.stereotype.Service;
import ua.com.alevel.persistence.crudhelper.CrudRepositoryHelper;
import ua.com.alevel.persistence.entity.educationalprocess.Report;
import ua.com.alevel.persistence.repository.educationalprocess.ReportRepository;
import ua.com.alevel.service.AbstractService;

@Service
public class ReportService extends AbstractService<Report, ReportRepository> {

    private final ReportRepository reportRepository;
    private final CrudRepositoryHelper<Report, ReportRepository> crudRepositoryHelper;

    public ReportService(ReportRepository reportRepository, CrudRepositoryHelper<Report, ReportRepository> crudRepositoryHelper) {
        super(reportRepository, crudRepositoryHelper);
        this.reportRepository = reportRepository;
        this.crudRepositoryHelper = crudRepositoryHelper;
    }
}

File: client-tutor/src/app/shared/services/educationalprocess/report.service.ts
import {Injectable} from '@angular/core';
import {AbstractService} from "../abstract.service";
import {HttpClient} from "@angular/common/http";
import {Report} from "../../models/educationalprocess/report.model";

@Injectable({
  providedIn: 'root'
})
export class ReportService extends AbstractService<Report> {

  override resourceUrl = '/api/reports';

  constructor(public override httpClient: HttpClient) {
    super(httpClient);
  }
}
